service-registry/src/test/java/io/servicecomb/serviceregistry/definition/TestMicroserviceDefinitionManager.java

```diff
@@ -0,0 +1,107 @@
+package io.servicecomb.serviceregistry.definition;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import io.servicecomb.config.archaius.sources.ConfigModel;
+import io.servicecomb.config.archaius.sources.MicroserviceConfigLoader;
+
+public class TestMicroserviceDefinitionManager {
+    private MicroserviceDefinitionManager createMicroserviceDefinitionManager(ConfigModel... configs) {
+        MicroserviceConfigLoader loader = new MicroserviceConfigLoader();
+        if (configs != null) {
+            for (ConfigModel config : configs) {
+                loader.getConfigModelList().add(config);
+            }
+        }
+        loader.loadAndSort();
+
+        MicroserviceDefinitionManager microserviceDefinitionManager = new MicroserviceDefinitionManager();
+        microserviceDefinitionManager.init(loader.getConfigModelList());
+
+        return microserviceDefinitionManager;
+    }
+
+    private ConfigModel createConfigModel(String app, String name) throws MalformedURLException {
+        ConfigModel config = new ConfigModel();
+        config.setUrl(new URL(String.format("jar:file:/file.jar!/%s/%s", app, name)));
+        config.setConfig(new HashMap<>());
+        config.getConfig().put(DefinitionConst.appIdKey, app);
+
+        Map<String, Object> map = new HashMap<>();
+        map.put(DefinitionConst.nameKey, name);
+        config.getConfig().put(DefinitionConst.serviceDescriptionKey, map);
+
+        return config;
+    }
+
+    @Test
+    public void testDefaultAppId() {
+        MicroserviceDefinitionManager microserviceDefinitionManager = createMicroserviceDefinitionManager();
+        Assert.assertEquals("default", microserviceDefinitionManager.getAppId());
+        Assert.assertEquals(new File(this.getClass().getClassLoader().getResource("").getPath()).getPath(),
+                microserviceDefinitionManager.getDefinitionMap().get("default").getRootPath());
+    }
+
+    @Test
+    public void testOneAppId() {
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        URL test1URL = classLoader.getResource("app1.yaml");
+        System.setProperty("cse.configurationSource.additionalUrls", test1URL.toString());
+
+        MicroserviceDefinitionManager microserviceDefinitionManager = createMicroserviceDefinitionManager();
+        Assert.assertEquals("app1", microserviceDefinitionManager.getAppId());
+
+        System.clearProperty("cse.configurationSource.additionalUrls");
+    }
+
+    @Test
+    public void testMultiAppId() throws MalformedURLException {
+        ConfigModel c1 = createConfigModel("app1", "ms1");
+        ConfigModel c2 = createConfigModel("app2", "ms2");
+
+        try {
+            createMicroserviceDefinitionManager(c1, c2);
+            Assert.assertEquals(1, 2);
```

Comment:
Assert.fail("Expect Exception xxx") can help us find out the test failed reason.